util/chunk/row_container.go

```diff
@@ -185,11 +212,11 @@ func (c *RowContainer) GetDiskTracker() *disk.Tracker {
 
 // Close close the RowContainer
 func (c *RowContainer) Close() (err error) {
-	if c.AlreadySpilled() {
-		err = c.recordsInDisk.Close()
-		c.recordsInDisk = nil
+	if c.alreadySpilled() {
```

Comment:
All the usage of `alreadySpilled` should be read locked because it touches `c.m.recordsInDisk`

Reply:
Because `SpillToDisk` may be called after Close happens, this situation should be considered.

Reply:
Ok, I add RLock in Close().